src/main/java/manufacturing/dao/jdbc/CarDaoJdbc.java

```diff
@@ -0,0 +1,201 @@
+import manufacturing.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbc implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+
+            insertStatement.setLong(1, car.getManufacturer().getId());
+            insertStatement.setString(2, car.getModel());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not insert " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.car_id AS car_id, cars.model, m.manufacturer_id AS m_id, "
+                + "m.manufacturer_name AS m_name, m.manufacturer_country AS m_country, "
+                + "d.driver_id AS d_id, d.driver_name AS d_name, d.license_number FROM cars "
+                + "LEFT JOIN cars_drivers cd ON cars.car_id = cd.car_id "
+                + "LEFT JOIN drivers d ON d.driver_id = cd.driver_id "
+                + "LEFT JOIN manufacturers m ON cars.manufacturer_id = m.manufacturer_id "
+                + "WHERE cars.car_id = ? AND cars.deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getByIdStatement = connection.prepareStatement(
+                        query,
+                        ResultSet.TYPE_SCROLL_INSENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+
+            getByIdStatement.setLong(1, id);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.first()) {
+                car = createCarInstance(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not get car by id " + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.car_id AS car_id, cars.model, m.manufacturer_id AS m_id, "
+                + "m.manufacturer_name AS m_name, m.manufacturer_country AS m_country, "
+                + "d.driver_id AS d_id, d.driver_name AS d_name, d.license_number FROM cars "
+                + "LEFT JOIN cars_drivers cd ON cars.car_id = cd.car_id "
+                + "LEFT JOIN drivers d ON d.driver_id = cd.driver_id "
+                + "LEFT JOIN manufacturers m ON cars.manufacturer_id = m.manufacturer_id "
+                + "WHERE cars.deleted = false";
```

Comment:
same!!! Don't repeat your self

Reply:
I could simplify the code to a certain extent, but it would harm the performance